spec/elastic_apm/metadata/cloud_info_spec.rb

```diff
@@ -0,0 +1,182 @@
+      subject { described_class.new(config) }
+
+      context 'aws' do
+        let(:config) { Config.new(cloud_provider: 'aws') }
+
+        before do
+          @aws_mock =
+            WebMock.stub_request(:get, Metadata::CloudInfo::AWS_URI)
+            .to_return(body: CloudExamples::AWS_EXAMPLE)
+        end
+
+        it 'fetches metadata from aws' do
+          subject.fetch!
+
+          expect(subject.account_id).to eq '123456789012'
+          expect(subject.instance_id).to eq 'i-1234567890abcdef0'
+          expect(subject.availability_zone).to eq 'us-west-2b'
+          expect(subject.provider).to eq 'aws'
+          expect(subject.region).to eq 'us-west-2'
+
+          expect(@aws_mock).to have_been_requested
+        end
+      end
+
+      context 'gcp' do
+        let(:config) { Config.new(cloud_provider: 'gcp') }
+
+        before do
+          @gcp_mock =
+            WebMock.stub_request(:get, Metadata::CloudInfo::GCP_URI)
+            .to_return(body: CloudExamples::GCP_EXAMPLE)
+        end
+
+        it 'fetches metadata from gcp' do
+          subject.fetch!
+
+          expect(subject.provider).to eq('gcp')
+          expect(subject.instance_id).to eq(4306570268266786072)
+          expect(subject.instance_name).to eq("basepi-test")
+          expect(subject.project_id).to eq(513326162531)
+          expect(subject.project_name).to eq("elastic-apm")
+          expect(subject.availability_zone).to eq('us-west3-a')
+          expect(subject.region).to eq('us-west3')
+          expect(subject.machine_type).to eq('projects/513326162531/machineTypes/n1-standard-1')
+
+          expect(@gcp_mock).to have_been_requested
+        end
+      end
+
+      context 'azure' do
+        let(:config) { Config.new(cloud_provider: 'azure') }
+
+        before do
+          @azure_mock =
+            WebMock.stub_request(:get, Metadata::CloudInfo::AZURE_URI)
+            .to_return(body: CloudExamples::AZURE_EXAMPLE)
+        end
+
+        it 'fetches metadata from azure' do
+          subject.fetch!
+
+          expect(subject.provider).to eq "azure"
+          expect(subject.account_id).to eq "7657426d-c4c3-44ac-88a2-3b2cd59e6dba"
+          expect(subject.instance_id).to eq "e11ebedc-019d-427f-84dd-56cd4388d3a8"
+          expect(subject.instance_name).to eq "basepi-test"
+          expect(subject.project_name).to eq "basepi-testing"
+          expect(subject.machine_type).to eq "Standard_D2s_v3"
+          expect(subject.region).to eq "westus2"
+
+          expect(@azure_mock).to have_been_requested
+        end
+      end
+
+      context 'auto' do
+        let(:config) { Config.new(cloud_provider: 'auto') }
+
+        context 'timeouts' do
+          it 'tries all three' do
+            WebMock.stub_request(:get, Metadata::CloudInfo::AWS_URI).to_timeout
+            WebMock.stub_request(:get, Metadata::CloudInfo::GCP_URI).to_timeout
+            WebMock.stub_request(:get, Metadata::CloudInfo::AZURE_URI).to_timeout
```

Comment:
cool, didn't know about the `to_timeout` expectation :)

Reply:
Me neither! Found it when writing this PR.